src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c

```diff
@@ -20,22 +22,154 @@ const ActorInit Bg_Lotus_InitVars = {
+        }
+    }
+    if (thisx->unk160 < thisx->dyna.actor.floorHeight) {
+        thisx->dyna.actor.world.pos.y = thisx->dyna.actor.floorHeight;
+        goto block_12;
+    }
+    thisx->dyna.actor.world.pos.y = thisx->unk160;
+    if (DynaPolyActor_IsPlayerOnTop((DynaPolyActor*)thisx) != 0) {
+        if (thisx->unk164 == 0) {
+            EffectSsGRipple_Spawn(play, &thisx->dyna.actor.world.pos, 0x3E8, 0x578, 0);
+            EffectSsGRipple_Spawn(play, &thisx->dyna.actor.world.pos, 0x3E8, 0x578, 8);
+            thisx->unk166 = 0x28;
+        }
+        if (gSaveContext.save.playerForm != 3) {
+            thisx->unk166 = 0x28;
+            thisx->dyna.actor.flags |= 0x10;
+            thisx->unk15C = func_80AD6A88;
+            return;
+        }
+        thisx->unk164 = 1U;
+        goto block_12;
+    }
+    thisx->unk164 = 0U;
+block_12:
+    if (thisx->unk166 > 0) {
+        thisx->unk166 = (thisx->unk166 - 1);
+    }
+    func_80AD6830(&thisx->dyna.actor);
+}
+
+void func_80AD6A88(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+
+    if (this->unk160 < thisx->world.pos.y) {
+        thisx->world.pos.y = this->unk160;
+    }
+    thisx->world.pos.y -= 1.0f;
+    if (thisx->world.pos.y <= thisx->floorHeight) {
+        thisx->world.pos.y = thisx->floorHeight;
+        this->unk166 = 0;
+    }
+    if (this->unk166 > 0) {
+        this->unk166 = (this->unk166 - 1);
+        func_80AD6830(thisx);
+        return;
+    }
+    if (Math_StepToF(&thisx->scale.x, 0.0f, 0.0050000004f) != 0) {
+        thisx->draw = (void*)0;
+        this->unk166 = 0x64;
+        DynaPoly_DisableCollision(play, &play->colCtx.dyna, this->dyna.bgId);
+        this->unk15C = func_80AD6B68;
+    }
+    thisx->scale.z = thisx->scale.x;
+}
+
+void func_80AD6B68(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+
+    f32 temp_fv0;
+    f32 temp_fv1;
+
+    if (this->unk166 > 0) {
+        this->unk166--;
+        return;
+    }
+    if ((thisx->xzDistToPlayer > 100.0f) && (thisx->projectedPos.z < 0.0f)) {
+        thisx->draw = BgLotus_Draw;
+        DynaPoly_EnableCollision(play, &play->colCtx.dyna, this->dyna.bgId);
+        Actor_SetScale(thisx, 0.1f);
+        temp_fv0 = this->unk160;
+        temp_fv1 = thisx->floorHeight;
+        if (temp_fv0 < temp_fv1) {
+            thisx->world.pos.y = temp_fv1;
+        } else {
+            thisx->world.pos.y = temp_fv0;
+        }
+        thisx->flags &= ~ACTOR_FLAG_10;
+        this->unk168 = 0x60;
+        this->unk15C = &func_80AD68DC;
+        thisx->world.pos.x = thisx->home.pos.x;
+        thisx->world.pos.z = thisx->home.pos.z;
+    }
+}
+
+void BgLotus_Update(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+
+    s32 pad;
+    WaterBox* sp2C;
+
+    WaterBox_GetSurface1_2(play, &play->colCtx, this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.z,
+                           &this->unk160, &sp2C);
+    ((UNK_TYPE(*)(BgLotus*, PlayState*))(this->unk15C))(this, play);
```

Comment:
After you change the type away from s32 (previous review), you can change this to a much more readable
```suggestion
    this->actionFunc(this, play);
```